### PR TITLE
Fix question listing and navigation in quiz reports

### LINKED ISSUE #12334: Quiz review - Missing selection effect, incorrect question titles
## Observed behavior

Observed while testing https://github.com/learningequality/kolibri/pull/12312 - When I attempt to review the results of a completed quiz I don't see properly which question I have selected to review and also when I select a question there's a mismatch between the selected question form the list and the actually displayed title of the question.

Video:

https://github.com/learningequality/kolibri/assets/79847249/7b5c0c31-507c-4774-afa4-c302654280a4

## Expected behavior

Clear indication which is the currently selected question and matching question title.

## Steps to reproduce the issue

1. Install the build from https://github.com/learningequality/kolibri/pull/12312
2. Observe the steps in the provided video.

## Usage Details
Windows 11, Ubuntu 22  - Chrome

### PR DESCRIPTION
**Pull request by rtibbles**
## Summary
* Annotate sections with start and end section numbers.
* Simplify attempt log listing by requiring sections, so only having a mobile and non-mobile condition
* Fix attempt log listing by including the section context in the question numbers.
* Remove background to show selection state.
* Update question number display to show the number globally for the quiz, rather than per section.
* Update learner quiz display to be correlated.
* Cleanup cruft in learner quiz display that was making our computed props over complicated and redundant.

## References
Fixes [#12334](https://github.com/learningequality/kolibri/issues/12334)

## Reviewer guidance
* Test multi-section quiz reports to ensure that selection state is properly shown.
* Test to ensure that navigation between items occurs properly.
* Regression test practice quiz reports for learners and coaches
* Regression test exercise reports for coaches
* Ensure that multi-section quiz engagement and question selection functions properly.

----

## Testing checklist

- [ ] Contributor has fully tested the PR manually
- [ ] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Comment by pcenov** [comment]
Hi @rtibbles, everything seems ok but noticed 2 issues that can potentially cause confusion:

1. When all the sections are collapsed there's no indication to which section the displayed question belongs so that the learner/coach might have to expand several sections before finding out. Perhaps there should be a selection effect to the section title as well:



2. There's a mismatch between the question title within the section panel and the question title as displayed when viewing the details of the question. For example when I select Question 4 from the second section, it's displayed as Question 1 to the right and that's so for every new section:

https://github.com/learningequality/kolibri/assets/79847249/ad695713-7f6a-4aa5-8c27-0f402ae69c77

**Comment by rtibbles** [comment]
Thanks @pcenov - definitely useful to add an indicator at the section level too - I will use the same background colour.

Will make sure that the titles match!

**Comment by rtibbles** [comment]
Hi @pcenov I have fixed both these issues, and moved to a consistent 'global across the quiz' question numbering.

To be completely consistent, I have also updated this for learners taking the quiz too, so additional testing there would be very useful.

**Review by pcenov — APPROVED** [review]
Thanks @rtibbles LGTM!

File: kolibri/core/assets/src/exams/utils.js [hunk 1]
```diff
@@ -254,3 +254,26 @@ export function getExamReport(examId, tryIndex = 0, questionNumber = 0, interact
...
+        endQuestionNumber: questions.length - 1,
+      },
+    ];
+  }
+  let startQuestionNumber = 0;
+  return sections.map(section => {
+    const annotatedSection = {
+      ...section,
+      startQuestionNumber,
+      endQuestionNumber: startQuestionNumber + section.questions.length - 1,
```
Comment (marcellamaki) [inline kolibri/core/assets/src/exams/utils.js:278]: non-blocking but I think adding a comment here about the numbering convention would make this section a bit more readable. (I realize it's a refactor from a different file)

**Review by marcellamaki — APPROVED** [review]
Holding the context for all of this is a bit of a challenge, but that's a factor of the complexity of some of this code. One non-blocking suggestion for an additional comment

**Pull request merged**